modules/dicom_archive/php/viewdetails.class.inc

```diff
@@ -208,16 +203,18 @@ class ViewDetails extends \NDB_Form
         $sliceThickness = $series['SliceThickness'];
 
         foreach ($this->protocols as $protocol) {
-            if ((! $protocol['TR_range']
-                || $this->_inRange($tr, $protocol['TR_range']))
-                && (! $protocol['TE_range']
-                || $this->_inRange($te, $protocol['TE_range']))
-                && (! $protocol['TI_range']
-                || $this->_inRange($ti, $protocol['TI_range']))
-                && (! $protocol['slice_thickness_range']
+            if (((! $protocol['TR_min'] && ! $protocol['TR_max'])
```

Comment:
@llevitis The expression `(! $protocol['TR_min'] && ! $protocol['TR_max'])` actually evaluates to `true` if `$protocol['TR_min'] == 0` and `$protocol['TR_max'] == 0`, so I think it is not the right test to perform. Basically it means that if you setup your `mri_protocol` table with a TR min and max set to 0, that TR constraint is basically useless as your algorithm always says that it is satisfied.

The `_in_range` function should handle all the possible min/max combinations, including those for which either min or max (or both) are null. But I notice that in this function if min and max are null, false is returned. Any particular reason?

Reply:
That's a good point regarding a null+null combination having to actually evaluate to true. However, given that the default for the min & max are null, I don't know if we should expect a user to add a value of 0 for the min & max. @ridz1208, thoughts?